go/mysql/mysql56_gtid_set.go

```diff
@@ -334,6 +334,83 @@ func (set Mysql56GTIDSet) AddGTID(gtid GTID) GTIDSet {
 	return newSet
 }
 
+// Union implements GTIDSet.Union().
+func (set Mysql56GTIDSet) Union(other GTIDSet) GTIDSet {
+	if set == nil || other == nil {
+		return set
+	}
+	mydbOther, ok := other.(Mysql56GTIDSet)
+	if !ok {
+		return set
+	}
+
+	// Make a copy and add the new GTID in the proper place.
+	// This function is not supposed to modify the original set.
+	newSet := make(Mysql56GTIDSet)
+
+	for OtherSID, otherIntervals := range mydbOther {
+		intervals, ok := set[OtherSID]
+
+		if !ok {
+			// No matching server id, so we must add it from other set.
+			newSet[OtherSID] = otherIntervals
+			continue
+		}
+
+		// Found server id match between sets, so now we need to add each interval.
+		s1 := intervals[:]
+		s2 := otherIntervals[:]
```

Comment:
These are already slices, so you shouldn't need to slice-ify them.

```suggestion
		s1 := intervals
		s2 := otherIntervals
```

Reply:
Ahh, ok, I didn't know if it would create a new pointer if I did this. Good to know it does.

Reply:
Done.